db-scheduler/src/main/java/com/github/kagkarlsson/scheduler/SchedulerClient.java

```diff
@@ -352,6 +380,26 @@ public <T> void reschedule(TaskInstanceId taskInstanceId, Instant newExecutionTi
       }
     }
 
+    @Override
+    public <T> void rescheduleOrCreate(TaskInstance<T> taskInstance, Instant executionTime) {
+      Optional<ScheduledExecution<Object>> executionObject =
+          this.getScheduledExecutions().stream()
+              .filter(e -> e.getTaskInstance().getId().equals(taskInstance.getId()))
+              .findFirst();
+      if (executionObject.isPresent()) {
+        this.reschedule(taskInstance, executionTime, taskInstance.getData());
+      } else {
+        this.scheduleIfNotExists(taskInstance, executionTime);
+      }
```

Comment:
As long as we do not lock the row, there will be the possibility of race-conditions. 

* Maybe log a warn if `scheduleIfNotExists` returns false
* Also, maybe make `reschedule` return the success-code and warn on false on that as well.

Reply:
In my opinion, scheduleIfNotExists should generally create a warning log entry in the event of failure. This is already the case. I therefore do not see the need to generate an additional log entry for this case. 

reschedule now also returns a boolean value and also creates a warning log entry in the event of an error. The only question here is how useful this is at the moment, as an ExecutionException is thrown in the only possible error case. It is therefore currently not possible for reschedule to return false, as an exception is thrown in this case.